src/app/loadout-builder/process-worker/process.ts

```diff
@@ -412,25 +259,22 @@ export function process(
               continue;
             }
 
-            const newArmorSet: IntermediateProcessArmorSet = {
-              armor,
-              stats,
-            };
-
             // Calculate the "tiers string" here, since most sets don't make it this far
             // A string version of the tier-level of each stat, must be lexically comparable
-            let tiers = '';
-            for (const statHash of statOrder) {
-              const value = stats[statHash];
-              const tier = statTier(value);
+            // TODO: It seems like constructing and comparing tiersString would be expensive but it's less so
+            // than comparing stat arrays element by element
+            let tiersString = '';
+            for (let index = 0; index < 6; index++) {
+              const value = Math.min(Math.max(stats[index], 0), 100);
+              const tier = tiers[index];
               // Make each stat exactly one code unit so the string compares correctly
-              const filter = statFilters[statHash];
+              const filter = statFiltersInStatOrder[index];
               if (!filter.ignored) {
-                tiers += tier.toString(11);
+                tiersString += tier.toString(16);
```

Comment:
Computers like powers of 2 - this was a noticeable speedup

Reply:
we live in hell

Reply:
Lets stick a comment on this so no one changes it